.github/workflows/lint.yml

```diff
@@ -79,16 +75,14 @@ jobs:
     name: render-defaults (${{ matrix.traits.name }})
     steps:
       - uses: actions/checkout@v4
-      - run: |
-          sudo apt-get install clang-format-14
       - uses: ./.github/actions/render/defaults
         id: render
         with:
           traits_name: ${{ matrix.traits.name }}
           library_name: ${{ matrix.traits.library }}
           library_url: ${{ matrix.traits.url }}
           disable_default_traits: ${{ matrix.traits.disable_pico }}
-      - run: clang-format-14 -i ${{ steps.render.outputs.file_path }}
+      - run: clang-format -i ${{ steps.render.outputs.file_path }}
```

Comment:
I'd prefer sticking to a specific version, because when the version changes it often changes how existing code is formatted. Though I agree that we should probably update it to a more recent version given clang18 is already released.
It also serves as documentation to allow contributors to use the same version before commiting.

Reply:
I suspect it was left over from me debugging so let's see if it passes (which it should)